baybe/searchspace/continuous.py

```diff
@@ -486,6 +486,11 @@ def sample_from_full_factorial(self, batch_size: int = 1) -> pd.DataFrame:
 
         return full_factorial.sample(n=batch_size).reset_index(drop=True)
 
+    @property
```

Comment:
See my longer comment about these properties below, but for this one in particular I'm not sure if it's really needed, since I can't imagine someone specifically asking a `ContinuousSubspace` about infos on its `TaskParameters`, a concept which doesn't exist in conti spaces. Since everyone would access this info via `SearchSpace`, I feel that this property is unnecessary?

Reply:
agree that taskparameter is not relevant here

But this was already put here with regards to consistency / the future: there could be various other mechanism that reduce the default kernel dimensions, e.g. if I specify that for 1 our of 4 conti parameters I want a special (parameter-specific) kernel, the effective default kernel dim would be 3 not 4, that would be reflected here.

Reply:
Yes, but I think that only confirms my argument: I attribute this decision to the surrogate, *not* to the searchspace. The searchspace is not aware how any of its components are to be handled and should thus not decide it. Different surrogates can apply arbitrarily different rules here. So in essence I suggest: if you want to extract the code into a functionality, it should belong the class making use of it / defining these rules. And that wouldn't even be the general surrogate class but only the *specific* surrogate classes that implement this particular split. Do you agree with this argument?

Thus, in your case, I could imagine it as a utility defined somewhere close to our current GP model, i.e. a simple function that takes the searchspace as input. But I'll leave it to you do decide if this improves the code or not.

Reply:
property removed